### PR TITLE
added Number of digits of a number in math

### PR DESCRIPTION
**Pull request by vicky1999**
### Created number_of_digits.cpp file under math directory.  The cpp program will calculate the number of digits in a number in two ways.  
One is by using normal approach of traversing through all the digits of the number.
The next approach is by using log10() function in math.h library.

### CODE REVIEW

**Comment by cclauss** [comment]
cpp_lint is failing the build...
```
math/number_of_digits.cpp:5:  Missing spaces around =  [whitespace/operators] [4]
math/number_of_digits.cpp:6:  Missing spaces around >  [whitespace/operators] [3]
math/number_of_digits.cpp:6:  Missing space before ( in while(  [whitespace/parens] [5]
math/number_of_digits.cpp:8:  Missing spaces around =  [whitespace/operators] [4]
math/number_of_digits.cpp:18:  Tab found; better to use spaces  [whitespace/tab] [1]
math/number_of_digits.cpp:18:  Missing spaces around =  [whitespace/operators] [4]
math/number_of_digits.cpp:19:  Tab found; better to use spaces  [whitespace/tab] [1]
math/number_of_digits.cpp:19:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
math/number_of_digits.cpp:20:  Tab found; better to use spaces  [whitespace/tab] [1]
math/number_of_digits.cpp:21:  Tab found; better to use spaces  [whitespace/tab] [1]
math/number_of_digits.cpp:21:  Should have a space between // and comment  [whitespace/comments] [4]
math/number_of_digits.cpp:22:  Tab found; better to use spaces  [whitespace/tab] [1]
math/number_of_digits.cpp:22:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
math/number_of_digits.cpp:23:  Tab found; better to use spaces  [whitespace/tab] [1]
Total errors found: 15
```

File: math/number_of_digits.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#include<iostream>
```
Comment (ayaankhan98) [inline math/number_of_digits.cpp:1]: modify this to `#include <iostream>`

File: math/number_of_digits.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#include<math.h>
+
+unsigned int digits(unsigned int n) {
+    int num_digits=0;
```
Comment (ayaankhan98) [inline math/number_of_digits.cpp:5]: modify this to `int num_digits = 0;`

File: math/number_of_digits.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    while(n>0) {
```
Comment (ayaankhan98) [inline math/number_of_digits.cpp:6]: modify this to `while (n > 0) {`

File: math/number_of_digits.cpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        num_digits++;
+        n=n/10;
```
Comment (ayaankhan98) [inline math/number_of_digits.cpp:8]: modify this to `n = n / 10;`

File: math/number_of_digits.cpp [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    }
+    return num_digits;
+}
+
+unsigned int digits_log(unsigned int n) {
+    return log10(n)+1;
+}
+
+int main() {
+	int num=12345;
```
Comment (ayaankhan98) [inline math/number_of_digits.cpp:18]: modify this to `int num = 12345;`

File: math/number_of_digits.cpp [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+	std::cout << "Number of digits in " << num << " is : " 
+		<< digits(num) << std::endl;
+	//Efficient approach to find the number of digits of a number
+	std::cout << "Number of digits in " << num << " is : " 
+		<< digits_log(num) << std::endl;
```
Comment (ayaankhan98) [inline math/number_of_digits.cpp:23]: `math/number_of_digits.cpp:23:  Tab found; better to use spaces  [whitespace/tab] [1]`

File: math/number_of_digits.cpp [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+unsigned int digits_log(unsigned int n) {
+    return log10(n)+1;
+}
+
+int main() {
+	int num=12345;
+	std::cout << "Number of digits in " << num << " is : " 
+		<< digits(num) << std::endl;
+	//Efficient approach to find the number of digits of a number
+	std::cout << "Number of digits in " << num << " is : "
```
Comment (ayaankhan98) [inline math/number_of_digits.cpp:22]: `math/number_of_digits.cpp:22:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]`

Comment (ayaankhan98) [reply math/number_of_digits.cpp:22]: `math/number_of_digits.cpp:22:  Tab found; better to use spaces  [whitespace/tab] [1]`

File: math/number_of_digits.cpp [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+unsigned int digits_log(unsigned int n) {
+    return log10(n)+1;
+}
+
+int main() {
+	int num=12345;
+	std::cout << "Number of digits in " << num << " is : " 
+		<< digits(num) << std::endl;
+	//Efficient approach to find the number of digits of a number
```
Comment (ayaankhan98) [inline math/number_of_digits.cpp:21]: `math/number_of_digits.cpp:21:  Tab found; better to use spaces  [whitespace/tab] [1]
math/number_of_digits.cpp:21:  Should have a space between // and comment  [whitespace/comments] [4]`

File: math/number_of_digits.cpp [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+unsigned int digits_log(unsigned int n) {
+    return log10(n)+1;
+}
+
+int main() {
+	int num=12345;
+	std::cout << "Number of digits in " << num << " is : " 
+		<< digits(num) << std::endl;
```
Comment (ayaankhan98) [inline math/number_of_digits.cpp:20]: `math/number_of_digits.cpp:20:  Tab found; better to use spaces  [whitespace/tab] [1]`

File: math/number_of_digits.cpp [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    return num_digits;
+}
+
+unsigned int digits_log(unsigned int n) {
+    return log10(n)+1;
+}
+
+int main() {
+	int num=12345;
+	std::cout << "Number of digits in " << num << " is : "
```
Comment (ayaankhan98) [inline math/number_of_digits.cpp:19]: `math/number_of_digits.cpp:19:  Tab found; better to use spaces  [whitespace/tab] [1]
math/number_of_digits.cpp:19:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]`

File: math/number_of_digits.cpp (hunk 4 above)
Comment (cclauss) [reply math/number_of_digits.cpp:8]: Or `n /= 10`

**Pull request closed without merging**